src/mococrw/padding_mode.h

```diff
@@ -0,0 +1,214 @@
+     *
+     * @return The hashing function
+     */
+    openssl::DigestTypes getHashingFunction() const
+    {
+        return _hashingFunction;
+    }
+
+private:
+    /**
+     * @brief The masking algorithm to be used. Not necessary for encryption, only when using the 
+     * signature facility.
+     */
+    openssl::DigestTypes _hashingFunction;
+};
+
+/**
+ * @brief PSSPadding
+ *
+ * This class defines the parameters specific to the RSA PSS padding mode.
+ */
+class PSSPadding: public RSAPadding {
+public:
+    PSSPadding(openssl::DigestTypes hashing = openssl::DigestTypes::SHA256,
+               openssl::DigestTypes masking = openssl::DigestTypes::SHA256,
+               int saltLength = c_defaultSaltLength)
+        : _hashingFunction{hashing}
+        , _maskingFunction(masking)
+        , _saltLength{saltLength}
+        {
+        }
+    openssl::RSAPaddingMode getPadding() const {
+        return openssl::RSAPaddingMode::PSS;
+    }
+
+private:
+    /**
+     * @brief The masking algorithm to be used. Not necessary for encryption.
+     */
+    openssl::DigestTypes _hashingFunction;
+
+    /**
+     * @brief The mgf1 to be used
+     */
+    openssl::DigestTypes _maskingFunction;
+
+    /**
+     * @brief The salt length
+     */
+    int _saltLength;
+
+    /**
+     * @brief Default value for the salt length
+     */
+    static const int c_defaultSaltLength = 20;
+};
+
+/**
+ * @brief OAEPPadding
+ *
+ * This class defines the parameters specific to the OAEP padding mode:
+ * - Hashing function
+ * - Masking function
+ * - Label
+ *
+ * All parameters have default values, and the label parameter is optional.
+ */
+class OAEPPadding: public RSAPadding {
+public:
+    OAEPPadding(openssl::DigestTypes hashing = openssl::DigestTypes::SHA256,
+                    openssl::DigestTypes masking = openssl::DigestTypes::SHA256,
+                    std::vector<uint8_t> label={})
+        : _hashingFunction{hashing}
+        , _maskingFunction{masking}
+        , _label{label}
+    {
+    }
+
+    /**
+     * @brief Get the hashing function
+     *
+     * Getter method for the hashing function.
+     *
+     * @return The hashing function
+     */
+    openssl::DigestTypes getHashingFunction() const
+    {
+        return _hashingFunction;
+    }
+
+    /**
+     * @brief Get the masking function
+     *
+     * Getter method for the masking function.
+     *
+     * @return The masking function
+     */
+    openssl::DigestTypes getMaskingFunction() const
+    {
+        return _maskingFunction;
+    }
+
+    /**
+     * @brief Get the OAEP label
+     *
+     * Getter method for the OAEP label.
+     *
+     * @return The OAEP label
+     */
+    std::vector<uint8_t> getLabel() const
+    {
+        return _label;
+    }
+
+    /**
+     * @brief Get the padding mode
+     *
+     * Getter method for the padding mode.
+     *
+     * @return \ref openssl::RSAPaddingMode::OAEP
+     */
+    openssl::RSAPaddingMode getPadding() const
```

Comment:
Please use the virtual keyword and override, if you are instantiating the virtual function "virtual openssl::RSAPaddingMode getPadding() const = 0;" from class RSAPadding.

Reply:
done